atlasdb-commons/src/main/java/com/palantir/common/time/NanoTime.java

```diff
@@ -14,26 +14,32 @@
  * limitations under the License.
  */
 
-package com.palantir.leader.lease;
+package com.palantir.common.time;
 
 import java.time.Duration;
 import java.util.Objects;
 import java.util.concurrent.locks.LockSupport;
 
-import com.google.common.annotations.VisibleForTesting;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
 
 public final class NanoTime implements Comparable<NanoTime> {
+    @JsonValue
     private final long time;
 
-    @VisibleForTesting
-    NanoTime(long time) {
+    @JsonCreator
+    private NanoTime(long time) {
         this.time = time;
     }
 
     public static NanoTime now() {
         return new NanoTime(System.nanoTime());
     }
 
+    public static NanoTime createForTests(long time) {
```

Comment:
*sniff*, what was wrong with the previous package-private constructor?

Reply:
useful to be accessed from outside the package, sadly. The alternative is a test dependency in the same package, but it's not like gradle/idea make this easy.

C++ friends would be nice here...